internal/handler/task.go

```diff
@@ -111,287 +175,121 @@ func (h *Handler) ListCmd() *cobra.Command {
 		Use:   "list",
 		Short: "List all tasks",
 		Run: func(cmd *cobra.Command, args []string) {
-			tasks, err := h.service.GetTasks()
+			t, err := h.service.GetTasks()
 			if err != nil {
-				fmt.Println("Error fetching tasks")
 				log.Fatal(err)
-				return
-			}
-
-			fmt.Println("Tasks:")
-			for i, task := range tasks {
-				str := "N/A"
-				if task.TS != nil {
-					str = task.TS.String()
-				}
-				fmt.Printf("%d. %s %s %s %d\n", i+1, task.Desc, str, task.Tags[0], task.Priority)
 			}
+			models.DisplayTasks(t)
 		},
 	}
-
 	return listCmd
 }
 
-func parseAdd(args []string) (*models.Task, error) {
-	var description string
-	var tags []string
-
-	if len(args) > 0 {
-		description = args[0]
-	} else {
-		return nil, errors.New("task name is required")
-	}
-
-	var date *time.Time
-	var timeStamp *models.TimeStamp
-	var priority *int
-
-	timeFlg := false
-	priorityFlg := false
-	for i := 1; i < len(args); i++ {
-		if args[i][0] == '+' {
-			tags = append(tags, args[i][1:])
-		} else if !timeFlg && args[i][0] == '@' { // this requires that the time expression be separated from '@' ex. gt add "work" @ 12-3 +MA
-			timeFlg = true
-			c := i + 3
-			j := i + 1
-			for ; j < len(args) && j < c; j++ {
-				t, ts, err := helper(args[j])
-				if err != nil && date == nil && timeStamp == nil {
-					return nil, err
-				} else if err != nil {
-					continue
-				}
-				if t != nil {
-					if date != nil {
-						return nil, errors.New("task date already set")
-					}
-					date = t
-				}
-				if ts != nil {
-					if timeStamp != nil {
-						return nil, errors.New("task timestamp already set")
-					}
-					timeStamp = ts
-				}
+func (h *Handler) DoneCmd() *cobra.Command {
+	doneCmd := &cobra.Command{
+		Use:   "done",
+		Short: "Marks task as complete by ID",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			ids, err := parseDone(args)
+			if err != nil {
+				log.Fatal(err)
 			}
-			i = j - 1
-		} else if !priorityFlg && args[i][0] == '%' {
-			priorityFlg = true
-			if len(args[i]) > 1 {
-				p, err := strconv.Atoi(args[i][1:])
+			for _, i := range ids {
+				t, err := h.service.GetTask(i)
 				if err != nil {
-					return nil, err
+					log.Fatal(err)
+				}
+				//check if task exists, because of nil pointer dereference
+				if t == nil {
```

Comment:
We have to check, if task is not nil, because it will always throw "panic: runtime error: invalid memory address or nil pointer dereference"